lib/src/disposable/disposable.dart

```diff
@@ -300,8 +300,20 @@ class Disposable implements _Disposable, DisposableManagerV3 {
   @override
   void manageStreamController(StreamController controller) {
     _throwOnInvalidCall('manageStreamController', 'controller', controller);
+    // If a single-subscription stream has a subscription and that
+    // subscription is subsequently canceled, the `done` future will
+    // complete, but there is no other way for us to tell that this
+    // is what has happened. If we then listen to the stream (since
+    // closing a stream that was never listen-to throws) we'll get
```

Comment:
Hmm, yeah, I think you're right. I thought it was throwing because when I wrote the tests (TDD FTW) it threw. But there were two actual problems and I was probably only hitting one of them in both tests.